editor/debugger/script_editor_debugger.cpp

```diff
@@ -471,7 +471,9 @@ void ScriptEditorDebugger::_parse_message(const String &p_msg, const Array &p_da
 	} else if (p_msg == "error") {
 		DebuggerMarshalls::OutputError oe;
 		ERR_FAIL_COND_MSG(oe.deserialize(p_data) == false, "Failed to deserialize error message");
-
+		if (!ProjectSettings::get_singleton()->get_setting("debug/gdscript/warnings/enable") && oe.warning) {
```

Comment:
```suggestion
		if (oe.warning && !ProjectSettings::get_singleton()->get_setting("debug/gdscript/warnings/enable")) {
```
Since checking for `oe.warning` is easier than other statement and should be made first.